vms/avm/service_test.go

```diff
@@ -32,16 +35,87 @@ func TestGetAssetDescription(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	s := &Service{vm: vm}
+	return genesisBytes, vm, s
+}
+
+func TestServiceIssueTx(t *testing.T) {
+	genesisBytes, vm, s := setup(t)
+	// TODO: Does this need to be here? Or can the lock be released once the
+	//       VM is initilized?
```

Comment:
In general, the lock needs to be held whenever a VM function is called by the consensus engine. Depending on the locking specified in `CreateHandlers` locks may be held when an RPC is called as well.

For the AVM specifically, the lock is assumed to be held both when an RPC is called and when any function is called, except for the ws API.